pulp_ansible/app/tasks/collections.py

```diff
@@ -608,13 +626,61 @@ async def _find_all_collections(self):
 
         await asyncio.gather(*tasks)
 
+    async def _can_be_optimized(self):
+        """Check last synced metadata time."""
+        root, api_version = await self._get_root_api(self.remote.url)
+        if api_version == 3:
+            downloader = self.remote.get_downloader(
+                url=root, silence_errors_for_response_status_codes={404}
+            )
+            try:
+                metadata = parse_metadata(await downloader.run())
+            except FileNotFoundError:
+                return False
+
+            sources = set()
+            if self.collection_info:
+                sources = {r.source for r in self.collection_info if r.source}
+            sources.add(self.remote.url)
+
+            published = metadata.get("published")
+            repo_metadata_time = self.repository.last_synced_metadata_time
+            repo_metadata_time = repo_metadata_time.isoformat() if repo_metadata_time else ""
+
+            # From serializers.DateTimeField.to_representation:
+            if repo_metadata_time.endswith("+00:00"):
+                repo_metadata_time = repo_metadata_time[:-6] + "Z"
```

Comment:
How can this part be made to have `published and `repo_metadata_time` both be DateTime objects instead of strings? I think we want to use DateTime for comparisons when those strings represent time.

Reply:
this is actually what this code is doing, converting the string it gets from serializer to datetime